src/main/scala/fi/oph/koski/valpas/ValpasKuntailmoitusService.scala

```diff
@@ -366,21 +372,32 @@ class ValpasKuntailmoitusService(
     tekijäOrganisaatio: Option[OrganisaatioWithOid],
     oppijanOppilaitokset: Set[ValpasOppilaitos.Oid]
   )(implicit session: ValpasSession): Either[HttpStatus, Set[OrganisaatioWithOid]] = {
-    tekijäOrganisaatio match {
-      // Jos organisaatio on annettu, ei palauteta toistaiseksi mitään muita vaihtoehtoja,
+    val sallitutRoolit = Set(
+      ValpasRooli.OPPILAITOS_HAKEUTUMINEN,
+      ValpasRooli.OPPILAITOS_SUORITTAMINEN,
+      ValpasRooli.KUNTA
+    )
+
+    val oidsToCheck = tekijäOrganisaatio match {
+      // Jos organisaatio on jo annettu, ei palauteta toistaiseksi mitään muita vaihtoehtoja,
       // vaikka jollekin yksittäiselle oppijalle voisikin lähettää ilmoituksen muusta oppilaitoksesta käsin.
-      case Some(o) if accessResolver.filterByOikeudet(ValpasRooli.OPPILAITOS_HAKEUTUMINEN)(Set(o.oid)) == Set(o.oid) => Right(Set(o))
-      case Some(o) => Right(Set.empty)
-      case None => {
-        val maybeOrganisaatiot = accessResolver.filterByOikeudet(ValpasRooli.OPPILAITOS_HAKEUTUMINEN)(oppijanOppilaitokset)
-          .map(oid => organisaatioRepository.getOrganisaatio(oid))
-
-        if (maybeOrganisaatiot.contains(None)) {
-          Left(ValpasErrorCategory.internalError("Kaikkia oppijan organisaatioita ei löydy organisaatiopalvelusta"))
-        } else {
-          Right(maybeOrganisaatiot.flatten)
-        }
-      }
+      case Some(o) => Set(o.oid)
+      case None => oppijanOppilaitokset
+    }
+
+    val organisaatiot = oidsToCheck
+      .filter(oid => sallitutRoolit.exists(rooli => accessResolver.accessToOrg(rooli, oid)))
+      .map(organisaatioRepository.getOrganisaatio)
+
+    val kunnat = organisaatioService.omatOrganisaatiotJaKayttooikeusroolit
```

Comment:
voisi olla selkeämpi, että kuntia ei palauteta, jos tekijäOrganisaatio on määritelty. tai vaihtoehtoisesti palautetaan kaikki oppilaitoksetkin (chekki yllä). Kommentoin tästä juuri flowdockissakin.

Reply:
Jäädäänkö odottamaan kommenttia?

Reply:
toivotaan, ettei toivota kuntana ilmoitusten tekomahdollisuutta listanäkymistä... hakenut muualle -esitiedot nimittäin käsitellään frontissa, joten ne toimivat väärin, jos tekijäksi valitaan joku muu kuin se organisaatio, minkä listanäkymässä ollaan. Ja siirtämisessä bäkkäriin olisi aika tavalla vaivaa.